### DISCUSSION TITLE
The server sends messages to all clients on a regular basis

### DISCUSSION THREAD

**Discussion by FreeVB** [jchristn/SuperSimpleTcp#202] [Q&A]
Hello, is there a way for the server to send messages to all clients on a regular basis?

It is an active send, for example, a message with a specified content is sent to all connected clients every 5 seconds.

**Comment by FreeVB** [comment]
https://github.com/jchristn/SuperSimpleTcp/issues/197
In addition, I found a solution to this:
_Server = new SimpleTcpServer("0.0.0.0", 1236)

**Comment by jchristn** [comment]
Hi @FreeVB you can use the server ```GetClients()``` method and then iterate, sending each a message.  Broadcast functionality is not built into the library.

**Comment by FreeVB** [comment]
Thanks, I've implemented it via the timer, and I'm looking forward to implementing the broadcast function in the library.

**Reply by jchristn** [reply]
Awesome @FreeVB !  Please let me know how it goes!

**Comment by FreeVB** [comment]
@jchristn  Here's the code I implemented with a timer to send server time to all clients every 3 seconds.
```
private void StartTimer()
{
    _timer = new System.Timers.Timer(3000);
    _timer.Elapsed += SentServerTime2ClientAsync;
    _timer.AutoReset = true;
    _timer.Start();
}
async void SentServerTime2ClientAsync(object sender, ElapsedEventArgs e)
{
    List<string> clients = _Server.GetClients().ToList();
    if (clients != null && clients.Count > 0)
    {
        foreach (string curr in clients)
        {
            _Server.SendAsync(curr, Encoding.UTF8.GetBytes(DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss"))).Wait();
        }
    }
}
async void StartServer()
{
    _Server = new SimpleTcpServer("0.0.0.0", 808);// 输入0.0.0.0可以实现监听当前服务器上所有可用IP地址
    _Server.Events.ClientConnected += ClientConnected;
    _Server.Events.ClientDisconnected += ClientDisconnected;
    _Server.Events.DataReceived += DataReceived;
    _Server.Events.DataSent += ServerDataSent;
    _Server.Keepalive.EnableTcpKeepAlives = true;
    _Server.Settings.IdleClientTimeoutMs = _IdleClientTimeoutMs;
    _Server.Settings.MutuallyAuthenticate = false;
    _Server.Settings.AcceptInvalidCertificates = true;
    _Server.Settings.NoDelay = true;
    _Server.Logger = PrintServeLog;
    _Server.Start();
    StartTimer();
    PrintServeLog("Tcp 服务已启动,IP 为：" + _Server.IpAddress + "，端口为 " + _Server.Port);
    this.uiButtonStartServe.Enabled = false;
    this.uiButtonEndServe.Enabled = true;
}
```

**Reply by jchristn** [reply]
Very nice!  Is it working okay for you?

**Comment by FreeVB** [comment]
Yes, I can achieve the results I want, but I don't know if anything unusual will happen if there are more clients connecting.
Such as blocking, crashing, etc.
